<launch>
    <arg name="mav_name" default="firefly"/>
    <arg name="frame_id" default="odom" />
    <arg name="verbose" default="false"/>
    <arg name="voxel_size" default="0.20" />
    <arg name="robot_radius" default="0.6" />
    <arg name="clear_radius" default="1.0" />

    <group ns="$(arg mav_name)">
        <node name="scoutair_planner" pkg="scoutair_planner" type="test_esdf_map_server_node" output="screen">
            <remap from="odometry" to="ground_truth/odometry" />
            <remap from="scoutair_planner/esdf_map_in" to="esdf_map" />
            <remap from="scoutair_planner/tsdf_map_in" to="tsdf_map" />

            <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
            <param name="tsdf_voxels_per_side" value="16" />
            <param name="esdf_max_distance_m" value="2.0" />
            <param name="update_mesh_every_n_sec" value="0.0" />
            <param name="traversability_radius" value="$(arg robot_radius)" />
            <param name="publish_traversable" value="true" />
            <param name="slice_level" value="1.0" />
            <param name="publish_slices" value="true" />

            <param name="local_frame_id" value="$(arg frame_id)" />
            <param name="world_frame" value="$(arg frame_id)" />
            <param name="replan_dt" value="0.25" />
            <param name="command_publishing_dt" value="0.25" />
            <param name="replan_lookahead_sec" value="1.0" />
            <param name="mpc_prediction_horizon" value="300" />

            <param name="robot_radius" value="$(arg robot_radius)" />
            <param name="planning_horizon_m" value="20.0" />
            <param name="autostart" value="true" />
            <param name="verbose" value="true" />
            <param name="v_max" value="2.0" />
            <param name="a_max" value="2.0" />
            <param name="avoid_collisions" value="true" />

            <!-- Perception utils -->
            <param name="perception_utils/top_angle" value="0.56125" type="double"/>;
            <param name="perception_utils/left_angle" value="0.69222" type="double"/>;
            <param name="perception_utils/right_angle" value="0.68901" type="double"/>;
            <param name="perception_utils/max_dist" value="4.5" type="double"/>;
            <param name="perception_utils/vis_dist" value="1.0" type="double"/>;

            <!-- What to do in case of failure. -->
            <param name="goal_selector_strategy" value="none" />
    </node>
    </group>
</launch>